### PR TITLE
Replace deprecated `huggingface_hub.Repository` by `HfApi` 

### PR DESCRIPTION
**Pull request by qgallouedec**
## Description

Copy of #462 

closes https://github.com/DLR-RM/rl-baselines3-zoo/issues/463

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: rl_zoo3/push_to_hub.py [hunk 1]
```diff
@@ -198,10 +198,9 @@ def package_to_hub(
 
     # Git pull
     repo_local_path = Path(local_repo_path) / repo_name
-    repo = Repository(repo_local_path, clone_from=repo_url)
-    repo.git_pull(rebase=True)
 
-    repo.lfs_track(["*.mp4"])
```
Comment (araffin) [inline rl_zoo3/push_to_hub.py:204]: no more distinction about what is tracked with git and git lfs?

**Comment by araffin** [comment]
For the failure on CI: https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/243/commits/03db09ef40acaeb08a7a453023e872c2b8f4906e

and for `ValueError: loaded state dict contains a parameter group that doesn't match the size of optimizer's group`

I hope it is not https://github.com/DLR-RM/stable-baselines3/pull/1975

File: rl_zoo3/push_to_hub.py [hunk 2]
```diff
@@ -195,13 +195,15 @@ def package_to_hub(
...
     repo_local_path = Path(local_repo_path) / repo_name
-    repo = Repository(repo_local_path, clone_from=repo_url)
-    repo.git_pull(rebase=True)
+    api.snapshot_download(repo_id=repo_id, local_dir=repo_local_path)
 
-    repo.lfs_track(["*.mp4"])
+    # Add mp4 files to .gitattributes
+    with open(repo_local_path / ".gitattributes", "a+") as f:
+        f.seek(0)  # Move the file pointer to the beginning of the file
+        if not any("*.mp4" in line for line in f):
```
Comment (araffin) [inline rl_zoo3/push_to_hub.py:205]: did you test that one?

Comment (qgallouedec) [reply rl_zoo3/push_to_hub.py:205]: Yes, any issue with this?

File: rl_zoo3/push_to_hub.py (hunk 1 above)
Comment (qgallouedec) [reply rl_zoo3/push_to_hub.py:204]: see L203

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Pull request merged**